eservice/docs/sservice.md

```diff
@@ -0,0 +1,94 @@
+* data-dir -- path where data files are stored
+* block-store -- name of the file where blocks are stored, defaults to ``${data}/${identity}.mdb``
+
+## Operations ##
+
+The storage service supports several operations for interacting with the block store.
+
+| URL           | Method | Request Body     | Response Body            | Description                                    |
+|:--------------|:-------|:-----------------|:-------------------------|------------------------------------------------|
+| /block/<id>   | GET    | None             | application/octet-string | Return the requested block                     |
+| /block/<id>   | PUT    | None             | None                     | Store the requested block                      |
+| /block/<id>   | HEAD   | None             | None                     | Return size of block or NOT_FOUND              |
+| /block/list   | GET    | None             | application/json         | List of all known blocks                       |
+| /block/status | POST   | application/json | application/json         | Size and expiration of requested blocks        |
+| /block/store  | POST   | multipart/form   | application/json         | Store multiple blocks, return proof of storage |
+| /info         | GET    | None             | application/json         | Request information about the service          |
+| /shutdown     | GET    | None             | None                     | Request to shutdown the service                |
+
+Across operations, a block id (the sha256 hash of the block) will be represented as a url-encoded, base64 string. Block data is always binary and identified as ``application/octet-string``.
+
+### Status ###
+
+The ``status`` operation returns the size and expiration time of any requested block that is currently maintained by the storage service. If a requested block is currently not managed by the storage service, length and expiration will be set to 0. Note that the expiration is the number of seconds in the future when the block will expire; it is not a wall clock time.
+
+#### Input ####
+
+```JSON
+[
+    "base64 encoded block hash", ...
+]
+```
+
+#### Output ####
+
+```JSON
+[
+    {
+        "block_id" : "base64 encoded block hash",
+        "size" : "integer",
+        "expiration" : "integer"
+    },
+    ...
+    {}
+]
+```
+### Store Blocks ###
+
+The ``store`` operation requests that the storage service manage a set of blocks for at least a requested interval of time. If the storage service agrees to manage the blocks for the requested time, it will sign the hash of the hashes of the stored blocks (computed in the same order as the blocks were requested). The request will be encoded as ``multipart/form``. The first section of the form will contain a ``JSON`` request that includes requested interval and a list of block identifiers. Each specified block will be in its own section in the form.
+
+#### Input ####
+
+
+```JSON
+{
+    "interval" : "integer",
```

Comment:
Future commit... i've changed this to expiration and clarified that it is positive integer number of seconds. The value still represents an interval since wall clock time is meaningless (and not helpful) in this context.

Reply:
i'd disagree that expiration time is really meaningless (in particular given that response must be expiration time and ultimately the TP will have to do time comparison) but it's true that interval can be more convenient.